sympy/ntheory/residue_ntheory.py

```diff
@@ -1550,11 +1668,15 @@ def discrete_log(n, a, b, order=None, prime_order=None):
     if order < 1000:
         return _discrete_log_trial_mul(n, a, b, order)
     elif prime_order:
-        if order < 1000000000000:
-            return _discrete_log_shanks_steps(n, a, b, order)
-        return _discrete_log_pollard_rho(n, a, b, order, order_factors)
-
-    return _discrete_log_pohlig_hellman(n, a, b, order)
+        # Shanks and Pollard rho are O(sqrt(order)) while index calculus is O(exp(2*sqrt(log(n)log(log(n)))))
+        # we compare the expected running times to determine the algorithmus which is expected to be faster
+        if 3*sqrt(log(n)*log(log(n))) < log(order):  # the number 3 was determined experimental
+            return _discrete_log_index_calculus(n, a, b, order)
+        elif order < 1000000000000:
+            return _discrete_log_shanks_steps(n, a, b, order)  # Shanks seems typically faster, but uses O(sqrt(order)) memory
```

Comment:
```suggestion
            # Shanks seems typically faster, but uses O(sqrt(order)) memory
            return _discrete_log_shanks_steps(n, a, b, order)
```

In [PEP 8](https://peps.python.org/pep-0008/#maximum-line-length), it is recommended that each line should contain no more than 79 characters. While SymPy does not strictly enforce this rule, it is advisable to adhere to it when possible.